_posts/2024-08-05-lessons-on-data-quality.md

```diff
@@ -0,0 +1,226 @@
+
+These findings were particularly surprising, because we expected that the state-of-the-art models, like GPT-4o would be able to produce code that was the most like the human-written code files, and hence would achieve similar Binoculars scores and be more difficult to identify.
+
+## Discovering a Problem
+
+Although these findings were interesting, they were also surprising, which meant we needed to exhibit caution. We decided to reexamine our process, starting with the data. It could be the case that we were seeing such good classification results because the quality our AI-written code was poor.
+
+After taking a closer look at our dataset, we found that this was indeed the case. There were a few noticeable issues. Firstly, the code we had scraped from GitHub contained a lot of short, config files which were polluting our dataset. There were also a lot of files with long licence and copyright statements.
+
+~~~python
+# pylint: disable=missing-docstring
+__all__ = (
+    1, # [invalid-all-object]
+    lambda: None, # [invalid-all-object]
+    None, # [invalid-all-object]
+)
+~~~
+
+_An example of a human-written Python config file from our dataset_
+
+Additionally, in the case of longer files, the LLMs were unable to capture all the functionality, so the resulting AI-written files were often filled with comments describing the omitted code.
+
+~~~java
+public class NestedFunctionCalls {
+
+    public void performQuery() {
+        equiJoinClause(
+            anyTree(
+                join("L_ORDERKEY", "O_ORDERKEY"),
+                join("P_PARTKEY", "L_PARTKEY")
+            ),
+            filter("LINEITEM_WITH_RETURNFLAG_TABLESCAN", "L_RETURNFLAG = 'R'"),
+            filter("ORDERS_WITH_SHIPPRIORITY_TABLESCAN", "O_SHIPPRIORITY >= 10")
+        );
+    }
+
+    public void equiJoinClause(Object... joins) {
+        // Implementation of equiJoinClause
+    }
+
+    public void anyTree(Object... nodes) {
+        // Implementation of anyTree
+    }
+
+    public void join(String table1, String table2) {
+        // Implementation of join
+    }
+~~~
+
+_An example of a code snippet, taken from an AI-written Java file from our dataset_
+
+## Back to the Drawing Board
+
+With the source of the issue being in our dataset, the obvious solution was to revisit our code generation pipeline.
+
+![jpg]({{ site.github.url }}/alaws/assets/data-quality/new-code-generation-pipeline.png)
+_Our new pipeline for generating AI code samples_
+
+First, we swapped our data source to use the [github-code-clean](https://huggingface.co/datasets/codeparrot/github-code-clean) dataset, in which the code files had been filtered to remove files that are auto-generated, have short line lengths, or a high proportion of non-alphanumeric characters.
```

Comment:
I think we need to justify why we thought this was ok despite this being a very likely training dataset.